File: bench/reversefile/reversefile.php
#!/usr/bin/php4 -f
<?php
/*
 $Id: reversefile.php,v 1.1 2004-05-28 22:25:05 bfulgham Exp $
 http://www.bagley.org/~doug/shootout/
*/
$fd = fopen("php://stdin", "r");
$lines = array();
while (!feof ($fd)) { array_push($lines, fgets($fd, 4096)); }
fclose($fd);
foreach (array_reverse($lines) as $line) print $line;
?>

File: bench/reversefile/reversefile.lua
#!/usr/bin/lua
-- $Id: reversefile.lua,v 1.1 2004-05-19 18:12:18 bfulgham Exp $
-- http://www.bagley.org/~doug/shootout/

local lines = {}
local w = io.write
nl = 0

gsub(read("*a"), "([^\n]+\n)", function (l)
    %lines[nl] = l
    nl = nl + 1
end)

nl = nl - 1
for i=nl,0,-1 do
    w(lines[i])
end
